plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java

```diff
@@ -148,8 +149,11 @@ private static org.apache.iceberg.types.Type toIcebergTypeInternal(Type type, Op
         if (type.equals(TIME_MICROS)) {
             return Types.TimeType.get();
         }
+        if (type.equals(TIMESTAMP_MILLIS)) {
+            return Types.TimestampType.withZone();
+        }
         if (type.equals(TIMESTAMP_MICROS)) {
-            return Types.TimestampType.withoutZone();
+            return Types.TimestampType.withZone();
```

Comment:
I'm not sure we should change this method, we don't want to map all Trino timestamps to iceberg's timestamp with timezone.
I think we should only have a change in `MigrateProcedure#toIcebergType` to map Trino timestamp to iceberg timestamp with timezone

Reply:
Agreed. Please don't change the behavior which is unrelated to migrate procedure.